src/Microsoft.Identity.Web/WebAppAuthenticationBuilderExtensions.cs

```diff
@@ -162,5 +165,41 @@ public static class WebAppAuthenticationBuilderExtensions
 
             return builder;
         }
+
+        internal /*for testing*/ static void ValidateRequiredOptions(MicrosoftIdentityOptions microsoftIdentityOptions)
+        {
+            string msg = "parameter cannot be null or whitespace, " +
+                    "and must be included in the configuration of the web app. " +
+                    "For instance, in the appsettings.json file. ";
+
+            if (string.IsNullOrEmpty(microsoftIdentityOptions.ClientId))
+            {
+                throw new ArgumentNullException(nameof(microsoftIdentityOptions.ClientId),
+                    string.Format(CultureInfo.InvariantCulture, "The ClientId " + msg));
+            }
+
+            if (string.IsNullOrEmpty(microsoftIdentityOptions.Instance))
+            {
+                throw new ArgumentNullException(nameof(microsoftIdentityOptions.Instance),
+                    string.Format(CultureInfo.InvariantCulture, "The Instance " + msg));
+            }
+
+            if (microsoftIdentityOptions.IsB2C)
+            {
+                if (string.IsNullOrEmpty(microsoftIdentityOptions.Domain))
+                {
+                    throw new ArgumentNullException(nameof(microsoftIdentityOptions.Domain),
+                        string.Format(CultureInfo.InvariantCulture, "The Domain " + msg));
+                }
+            }
+            else
+            {
+                if (string.IsNullOrEmpty(microsoftIdentityOptions.TenantId))
```

Comment:
What about ADFS?

Reply:
That's a good question @bgavrilMS 
For the moment, I did not even considered that Microsoft.Identity.Web coud support ADFS, but maybe it should.